packages/amplify-category-notifications/lib/multi-env-manager.js

```diff
@@ -183,87 +184,92 @@ async function pushChanges(context, pinpointNotificationsMeta) {
 }
 
 function writeData(context) {
-  writeAmplifyMeta(context);
-  writeMultienvData(context);
-}
-
-function writeMultienvData(context) {
   const categoryMeta = context.exeInfo.amplifyMeta[constants.CategoryName];
-  if (!categoryMeta) {
-    return;
-  }
-
-  const { envName } = context.exeInfo.localEnvInfo;
-
-  const availableChannels = notificationManager.getAvailableChannels();
   let pinpointMeta;
-  const enabledChannels = [];
-  const services = Object.keys(categoryMeta);
-  for (let i = 0; i < services.length; i++) {
-    const serviceMeta = categoryMeta[services[i]];
-    if (serviceMeta.service === constants.PinpointName &&
-                                serviceMeta.output &&
-                                serviceMeta.output.Id) {
-      availableChannels.forEach((channel) => {
-        if (serviceMeta.output[channel] && serviceMeta.output[channel].Enabled) {
-          enabledChannels.push(channel);
-        }
-      });
-      pinpointMeta = {
-        serviceName: services[i],
-        service: serviceMeta.service,
-        channels: enabledChannels,
-        Name: serviceMeta.output.Name,
-        Id: serviceMeta.output.Id,
-        Region: serviceMeta.output.Region,
-      };
-      break;
+  if (categoryMeta) {
+    const availableChannels = notificationManager.getAvailableChannels();
+    const enabledChannels = [];
+    const services = Object.keys(categoryMeta);
+    for (let i = 0; i < services.length; i++) {
+      const serviceMeta = categoryMeta[services[i]];
+      if (serviceMeta.service === constants.PinpointName &&
+                                  serviceMeta.output &&
+                                  serviceMeta.output.Id) {
+        availableChannels.forEach((channel) => {
+          if (serviceMeta.output[channel] && serviceMeta.output[channel].Enabled) {
+            enabledChannels.push(channel);
+          }
+        });
+        pinpointMeta = {
+          serviceName: services[i],
+          service: serviceMeta.service,
+          channels: enabledChannels,
+          Name: serviceMeta.output.Name,
+          Id: serviceMeta.output.Id,
+          Region: serviceMeta.output.Region,
+        };
+        break;
+      }
     }
   }
+  writeTeamProviderInfo(pinpointMeta, context);
+  writeBackendConfig(pinpointMeta, context.amplify.pathManager.getBackendConfigFilePath());
```

Comment:
Can we use this method from `amplify-cli` package.

Reply:
no, notifications need to update to both backend and current backend, it's a special workflow that currently no other categories use.

Reply:
That might be true, but I don't think serialization of file and writing it to disk should happen here